core/storage/suggestion/gae_models_test.py

```diff
@@ -0,0 +1,209 @@
+            suggestion_models.TARGET_TYPE_EXPLORATION,
+            self.target_id, 'thread_6')
+
+        observed_suggestion = suggestion_models.SuggestionModel.get_by_id(
+            suggestion_id)
+
+
+        self.assertEqual(
+            observed_suggestion.suggestion_type,
+            suggestion_models.SUGGESTION_EDIT_STATE_CONTENT)
+        self.assertEqual(
+            observed_suggestion.target_type,
+            suggestion_models.TARGET_TYPE_EXPLORATION)
+        self.assertEqual(
+            observed_suggestion.target_id, self.target_id)
+        self.assertEqual(
+            observed_suggestion.target_version_at_submission,
+            self.target_version_at_submission)
+        self.assertEqual(
+            observed_suggestion.status, suggestion_models.STATUS_IN_REVIEW)
+        self.assertEqual(observed_suggestion.author_id, 'author_3')
+        self.assertEqual(observed_suggestion.assigned_reviewer_id, 'reviewer_3')
+        self.assertEqual(observed_suggestion.reviewer_id, 'reviewer_3')
+        self.assertEqual(
+            observed_suggestion.score_category, self.score_category)
+        self.assertEqual(observed_suggestion.change_cmd, self.change_cmd)
+
+    def test_fail_create_object(self):
+        with self.assertRaisesRegexp(
+            Exception, 'There is already a suggestion with the given id: '
+                       'exploration.exp1.thread_1'):
+            suggestion_models.SuggestionModel.create(
+                suggestion_models.SUGGESTION_EDIT_STATE_CONTENT,
+                suggestion_models.TARGET_TYPE_EXPLORATION,
+                self.target_id, self.target_version_at_submission,
+                suggestion_models.STATUS_IN_REVIEW, 'author_3',
+                'reviewer_3', 'reviewer_3', self.change_cmd,
+                self.score_category, 'thread_1')
+
+    def test_get_suggestions_by_type(self):
+        self.assertEqual(
+            len(suggestion_models.SuggestionModel.get_suggestions_by_type(
+                suggestion_models.SUGGESTION_EDIT_STATE_CONTENT)), 5)
+        with self.assertRaisesRegexp(
+            Exception, 'Value \'invalid_suggestion_type\' for property'
+                       ' suggestion_type is not an allowed choice'):
+            suggestion_models.SuggestionModel.get_suggestions_by_type(
+                'invalid_suggestion_type')
+
+    def test_get_suggestion_by_author(self):
+        self.assertEqual(
+            len(suggestion_models.SuggestionModel.get_suggestions_by_author(
+                'author_1')), 1)
+        self.assertEqual(
+            len(suggestion_models.SuggestionModel.get_suggestions_by_author(
+                'author_2')), 3)
+        self.assertEqual(
+            len(suggestion_models.SuggestionModel.get_suggestions_by_author(
+                'author_3')), 1)
+        self.assertEqual(
+            len(suggestion_models.SuggestionModel.get_suggestions_by_author(
+                'author_invalid')), 0)
+
+
```

Comment:
Have just one line between method defs, ditto below

(/cc @apb7 to ensure we're tracking this for inclusion into the linting process)

Reply:
Done.